### ISSUE TITLE
LiveServer is not terminated cleanly which breaks pytest-cov functionality

### ISSUE THREAD

**Issue by jadkik** [pytest-dev/pytest-flask#49]
When using the `live_server` with `pytest-cov`, the coverage is not reported correctly. As mentioned in the issues pytest-dev/pytest-cov#44 and jarus/flask-testing#70, the problem is that the `live_server` process is not being terminated cleanly.

One of the proposed solutions in the flask-testing project works well here too:

```
def _clean_liveserver_stop(self):
    """Stop application process."""
    if self._process:
        try:
            os.kill(self._process.pid, signal.SIGINT)
            self._process.join(LIVESERVER_TIMEOUT)
        except Exception as ex:
            logging.error('Failed to join the live server process: %r', ex)
        finally:
            if self._process.is_alive():
                # If it's still alive, kill it
                self._process.terminate()
```

The problem is that such a solution is not cross-platform. How about adding a config option that enables/disables cleanly terminating the live server? Or always trying to cleanly exit the live server process and warn on platforms which do not support that easily/directly/without too much work?

I can submit a pull request if needed.

**Comment by vitalk** [comment]
Thank you for reporting this. PR is highly appreciated.

**Comment by jadkik** [comment]
Note that the pull request was not tested on Windows. The whole live_server does not work on Windows (Python 2.7), so I didn't bother. In case it's fixed, it is supposed to terminate it "non-cleanly".

**Comment by northernSage** [comment]
Seem to have been fixed with #53 so I'll be closing this.
